framework/src/modules/interoperability/sidechain/commands/submit_sidechain_cross_chain_update.ts

```diff
@@ -40,11 +41,14 @@ export class SubmitSidechainCrossChainUpdateCommand extends BaseCrossChainUpdate
 	public async execute(
 		context: CommandExecuteContext<CrossChainUpdateTransactionParams>,
 	): Promise<void> {
-		const [decodedCCMs, ok] = await this.executeCommon(context, false);
+		const { params } = context;
+
+		await this.verifyCertificateSignatureAndPartnerChainOutboxRoot(context);
+
+		const [decodedCCMs, ok] = await this.beforeCrossChainMessagesExecution(context, false);
 		if (!ok) {
 			return;
 		}
-		const { params } = context;
 
 		try {
```

Comment:
Why do we need this try now? It is just a security measure? And if we want to keep it, should we maybe move it to just before `this.apply` call?

Reply:
this try is for `finally` block where we delete `CONTEXT_STORE_KEY_CCM_PROCESSING` which makes sure that we always delete it doesn't matter if the tx fails or passes.